src/OrchardCore.Modules/OrchardCore.Sms.Azure/Services/AzureSmsProviderBase.cs

```diff
@@ -55,8 +57,9 @@ public virtual async Task<SmsResult> SendAsync(SmsMessage message)
 
         try
         {
-            var client = new SmsClient(_providerOptions.ConnectionString);
-            var response = await client.SendAsync(_providerOptions.PhoneNumber, message.To, message.Body);
+            _smsClient ??= new SmsClient(_providerOptions.ConnectionString);
```

Comment:
I don't think this change is useful unless the provider is a singleton, also if the client breaks for any reason it will be problematic

Reply:
It's useful to allow you to create one client for the entire request. You could have a request that sends 100 SMS messages, in this case we create one instance instead of 100. This is just a small optimization which should have no impact on the code logic. The client will be disposed along with the provider at the end of the request

Reply:
Got it